compiler/opam

```diff
@@ -1,5 +1,6 @@
 opam-version: "2.0"
 name: "jasmin"
+version: "dev"
```

Comment:
Can you please elaborate on why this change is important?

Should a similar patch be applied to the `opam` file for `mathcomp-word`?

Reply:
Actually, it's more that I wanted to apply this patch to `mathcomp-word` and decided to apply it here too. I think it's better practice in general that when you pin the git repo, you get "dev" as version name.